sqlglot/dialects/bigquery.py

```diff
@@ -480,7 +480,11 @@ def _parse_table_part(self, schema: bool = False) -> t.Optional[exp.Expression]:
                 table_name = this.name
                 while self._match(TokenType.DASH, advance=False) and self._next:
                     text = ""
-                    while self._curr and self._curr.token_type != TokenType.DOT:
+                    while (
+                        self._curr
```

Comment:
you don't need the self._curr check because self._is_connected also checks self._curr